crates/bevy_ecs/src/event.rs

```diff
@@ -798,6 +810,37 @@ impl<'a, E: Event> ExactSizeIterator for EventIteratorWithId<'a, E> {
     }
 }
 
+/// A registry of all of the [`Events`] in the [`World`], used by [`event_update_system`]
+/// to update all of the events.
+#[derive(Resource, Default)]
+pub struct EventRegistry {
+    event_updates: Vec<(ComponentId, fn(ComponentId, &mut World))>,
+}
+
+impl EventRegistry {
+    /// Registers an event type to be updated.
+    pub fn register_event<T: Event>(world: &mut World) {
+        let component_id = world.init_resource::<Events<T>>();
+        let mut registry = world.get_resource_or_insert_with(Self::default);
+        registry.event_updates.push((component_id, |id, world| {
+            // Bypass the type ID -> Component ID lookup with the cached component ID.
+            if let Some(ptr) = world.get_resource_mut_by_id(id) {
+                // SAFETY: The component was initialized with the type Events<T>.
```

Comment:
This comment (and the `ComponentId` type) is a bit confusing, as it refers to resources as components.

Reply:
Switched it to use `resource` instead.